src/index.js

```diff
@@ -25,7 +26,9 @@ ReactDOM.render(
             <WatchLaterContextPro>
               <LikeVideoContext>
                 <HistoryContextProvider>
+                  <AuthContextProvider>
                   <App />
+                  </AuthContextProvider>
```

Comment:
maybe you should consider placing auth provider at the top...below the browser router

Reply:
ok